src/ts/utils/roam.ts

```diff
@@ -58,7 +63,80 @@ export const Roam = {
         await Keyboard.pressEnter(15);
         await Keyboard.pressEnter();
         document.execCommand('paste')
+    },
+
+    moveCursorToStart() {
+        this.applyToCurrent(
+            node =>
+                new RoamNode(
+                    node.text,
+                    // new Selection(node.text.length * 2, node.text.length * 2)
+                    new Selection(0,0)
+                ))
+    },
+
+    moveCursorToEnd() {
+        this.applyToCurrent(
+            node =>
+                new RoamNode(
+                    node.text,
+                    new Selection(node.text.length * 2, node.text.length * 2)
+                ))
```

Comment:
this should probably be a method on RoamNode (and the one above as-well)

Reply:
By doing basically what you're doing here & returning a new object with updated values. Something like:

```js
withCursorAtTheEnd() { //name is subject to improvement
  return new RoamNode(
                    this.text,
                    new Selection(this.text.length * 2, this.text.length * 2)
                )
}
```

btw, I'm confused on why you have *2 here. It seems to me it should be just `.length`

Reply:
OK, done that, but I'm still a bit confused with this immutable thing...


and yeah *2 was from old code, not needed.